dragonfly/block/fire.go

```diff
@@ -0,0 +1,257 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/block/fire"
+	"github.com/df-mc/dragonfly/dragonfly/block/wood"
+	"github.com/df-mc/dragonfly/dragonfly/entity"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/df-mc/dragonfly/dragonfly/world/difficulty"
+	"math/rand"
+	"time"
+)
+
+// Fire is a non-solid block that can spread to nearby flammable blocks.
+type Fire struct {
+	noNBT
+	replaceable
+	transparent
+	empty
+
+	// Type is the type of fire.
+	Type fire.Fire
+	// Age affects how fire extinguishes. Newly placed fire starts at 0 and the value has a 1/3 chance of incrementing
+	// each block tick.
+	Age int
+}
+
+// FlammableBlock returns true if a block is flammable.
+func FlammableBlock(block world.Block) bool {
+	if flammable, ok := block.(Flammable); ok && flammable.FlammabilityInfo().Encouragement > 0 {
+		return true
+	}
+	return false
+}
+
+// woodTypeFlammable returns whether a type of wood is flammable.
+func woodTypeFlammable(w wood.Wood) bool {
```

Comment:
Might be better to move this into the wood.Wood an have it return the flammability info itself.